src/routes/bounce_activities_page.jsx

```diff
@@ -0,0 +1,227 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import BounceRulesActivity from "../components/BounceActivityContainer";
+import { getActivityLog } from "../utils/ruleCalls";
+
+export default class BounceActivityPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchCategory: "Bounce Action",
+      isBounceRulesTab: false,
+      isActivityLogTab: true,
+      searchToken: "",
+      isRedirectingToDetail: false,
+      selectedActivity: {},
+      activityLog: [],
+      currentPageIndex: 1,
+      rulesToShow: 10,
+      pagesToDisplay: 5,
+      filterOptions: [],
+      isValidFilter: true,
+      isFetching: true,
+    };
+    this.logout = this.logout.bind(this);
+    this.updateSearchToken = this.updateSearchToken.bind(this);
+    this.updateSearchCategory = this.updateSearchCategory.bind(this);
+    this.addFilter = this.addFilter.bind(this);
+    this.removeFilter = this.removeFilter.bind(this);
+    this.handleModalClose = this.handleModalClose.bind(this);
+    this.handleActivityTabClicked = this.handleActivityTabClicked.bind(this);
+    this.handleBounceTabClicked = this.handleBounceTabClicked.bind(this);
+    this.updatePageIndex = this.updatePageIndex.bind(this);
+    this.handleActivityClicked = this.handleActivityClicked.bind(this);
+    this.handlePrevClicked = this.handlePrevClicked.bind(this);
+    this.handleNextClicked = this.handleNextClicked.bind(this);
+  }
+
+  async componentDidMount() {
+    const { data: activities } = await getActivityLog();
+    if (activities) {
+      this.setState({
+        isFetching: false,
+        activityLog: activities.reverse(),
+      });
+    }
+  }
+
+  logout() {
+    const { history } = this.props;
+    localStorage.clear();
+    history.push("/");
+  }
+
+  handleModalClose(e) {
+    const { id } = e.currentTarget;
+    this.setState({
+      [id]: false,
+      isInvalidInput: false,
+      selectedRule: {},
+    });
+  }
+
+  updateSearchToken(e) {
+    this.setState({
+      searchToken: e.target.value.toLowerCase(),
+    });
+  }
+
+  updateSearchCategory(e) {
+    this.setState({
+      searchCategory: e.value.toLowerCase(),
+    });
+  }
+
+  filterRules(rules) {
+    const { searchToken } = this.state;
+    return rules.filter(
+      rule =>
+        rule.bounce_action.toLowerCase().includes(searchToken) ||
```

Comment:
To be safe, we should also toLowerCase() the searchToken so if it includes uppercase/lowercase it will match the rule's lowercase description or bounceaction